### ISSUE TITLE
Image Thumbnail in Laravel-elfinder

### ISSUE THREAD

**Issue by tskkh** [barryvdh/laravel-elfinder#97]
Hello all, I'm getting problem with the image thumbnail in laravel-elfinder integrate with CKeditor. It doesn't display the image thumbnail within the file browser display.

Anyone experienced with this please share me the solution.
 
![screenshot](https://cloud.githubusercontent.com/assets/8650616/9808012/00a47846-5885-11e5-9a6e-5c9977e7ecf0.png)

**Comment by janokary** [comment]
I have the same problem.

**Comment by barryvdh** [comment]
Does it try to load an image somewhere? (See js console)

**Comment by tskkh** [comment]
@barryvdh 
Yes, upload image from outside project.

**Comment by mln-mln-mln** [comment]
@barryvdh yes, its loading the thumbnail from the correct path ( it looks inside .tmb folder) but the thumbnails arent generated.

**Comment by barryvdh** [comment]
It could be that your Apache config is block folders/files starting with dots (try different folder name or change your htaccess file)
It could also be that the permissions are not correct.

**Comment by mln-mln-mln** [comment]
I tried a different folder, but it still fails to generate thumbnails. Is there something I have to do to generate the thumbnails in the tmb folder?

**Comment by mln-mln-mln** [comment]
The problem was, tmbURL wasn't set correctly. It's working perfectly now. :)

**Comment by JulioCesarSiqueira** [comment]
@mln-mln-mln  Where and how can i config the tmbURL ? Should be in config/elfinder.php ?

**Comment by nhtua** [comment]
@tskkh try setting the upload directory endpoint in both of path and URL are the same.
In my case, I use the dir has name "upload".

Change roots in config/elfinder.php
`'roots' => array(
        array(
            'driver' => 'LocalFileSystem',
            'path'   => env('UPLOAD_PATH', __DIR__.'/../public_html/upload'),
            'URL'    => env('UPLOAD_URL','/upload'),
            'attributes' => array(
                array( 
                    'pattern' => '/\/\./', //You can also set permissions for file types by adding, for example, <b>.jpg</b> inside pattern.
                    'read' => false,
                    'write' => false,
                    'locked' => true,
                    'hidden' => true
                )
            )
        )
    ),`

**Comment by ShekharKokate** [comment]
I have used type as browse. I am able to select the image but not able to upload.
Please help

**Comment by ojha007** [comment]
i used aws how to show preview image and add pagination

**Comment by ShekharKokate** [comment]
Can you please suggest, how to use pagination in elfinder?
I am getting issue to load all the images in elfinder. As the images are uploaded on amazon s3 and it is taking time to load all the images. If there will be any solution, please suggest.
